src/components/Attachment/Gallery.js

```diff
@@ -1,294 +1,240 @@
               // Until we do, lets disable this feature. saveToLocalByLongPress prop basically
               // opens up popup menu to with an option "Save to the album", which basically does nothing.
-              saveToLocalByLongPress={false}
-              onCancel={() => {
-                this.setState({ viewerModalOpen: false });
-              }}
-              index={this.state.viewerModalImageIndex}
               enableSwipeDown
+              imageUrls={galleryImages}
+              onCancel={() => setViewerModalOpen(false)}
               renderHeader={() => (
                 <GalleryHeader
-                  handleDismiss={() => {
-                    this.setState({ viewerModalOpen: false });
-                  }}
+                  handleDismiss={() => setViewerModalOpen(false)}
                 />
               )}
+              saveToLocalByLongPress={false}
             />
           </SafeAreaView>
         </Modal>
-      </React.Fragment>
+      </>
     );
   }
-}
-
-const HeaderContainer = styled.View`
-  display: flex;
-  flex-direction: row;
-  justify-content: flex-end;
-  position: absolute;
-  width: 100%;
-  z-index: 1000;
-  ${({ theme }) => theme.message.gallery.header.container.css}
-`;
 
-const HeaderButton = styled.TouchableOpacity`
-  width: 30;
-  height: 30;
-  margin-right: 20;
-  margin-top: 20;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  border-radius: 20;
-  ${({ theme }) => theme.message.gallery.header.button.css}
-`;
-
-const GalleryHeader = ({ handleDismiss }) => (
-  <HeaderContainer>
-    <HeaderButton onPress={handleDismiss}>
-      <CloseButton />
-    </HeaderButton>
-  </HeaderContainer>
-);
-
-export default withTranslationContext(
-  withMessageContentContext(themed(Gallery)),
-);
+  return (
+    <>
+      <GalleryContainer alignment={alignment} length={galleryImages.length}>
+        {galleryImages.slice(0, 4).map((image, i) => (
+          <ImageContainer
+            activeOpacity={0.8}
+            key={`gallery-item-${i}`}
+            length={galleryImages.length}
+            onLongPress={onLongPress}
+            onPress={() => {
+              setViewerModalOpen(true);
+              setViewerModalImageIndex(i);
+            }}
+            {...additionalTouchableProps}
+          >
+            {i === 3 && galleryImages.length > 4 ? (
+              <View style={{ width: '100%', height: '100%' }}>
+                <Image
+                  resizeMode='cover'
+                  source={{ uri: galleryImages[3].url }}
+                  style={{ width: '100%', height: '100%' }}
+                />
+                <View
+                  style={{
+                    alignItems: 'center',
+                    backgroundColor: 'rgba(0,0,0,0.69)',
+                    height: '100%',
+                    justifyContent: 'center',
+                    position: 'absolute',
+                    width: '100%',
```

Comment:
if this is supposed to fill the whole view does `flex: 1` not work, or `position: absolute` with top/left/right/bottom all set to 0? that's the more traditional setup with position absolute in my experience